crates/bpf-filtering/src/initializer.rs

```diff
@@ -157,9 +184,37 @@ impl Initializer {
         if decision.interesting {
             log::debug!("tracking {} {}", process.pid, process.image);
         }
-        log::trace!("{}: {}", process.pid, decision.as_raw());
-        self.cache.insert(process.pid, decision);
-        self.interest_map.set(process.pid, decision)?;
+        self.set_policy(process.pid, decision)
+    }
+
+    fn set_policy(&mut self, pid: Pid, policy: PolicyDecision) -> Result<()> {
+        log::trace!("{}: {}", pid, policy.as_raw());
+        self.cache.insert(pid, policy);
+        self.interest_map.set(pid, policy)
+    }
+
+    fn track_target_cgroups(&mut self) -> Result<()> {
+        let cgroups: Vec<String> = self
+            .config
+            .cgroup_targets
+            .iter()
+            // the cgroup.procfs file contains the list of pids belonging to this cgroup
+            .map(|cgroup| format!("/sys/fs/cgroup{cgroup}/cgroup.procs"))
+            .collect();
+        for cgroup in cgroups {
+            let processes = std::fs::read_to_string(&cgroup)
```

Comment:
noticed the `std::instead` of `tokio::fs`, is it intentional ?

Reply:
Ops! Didn't notice, I guess it was temporary code I wrote as a test during development and it slipped through. Nice catch. I also fixed the `std::fs::read_link` below